webapp/src/package.ts

```diff
@@ -39,11 +39,15 @@ export class File implements pxt.editor.IFile {
         return this.epkg.getPkgId() + "/" + this.name
     }
 
-    getTypeScriptName() {
+    getTextFileName() {
         if (this.epkg.isTopLevel()) return this.name
         else return "pxt_modules/" + this.epkg.getPkgId() + "/" + this.name
     }
 
+    getFileNameWithExtension(ext: string) {
+        return this.getTextFileName().replace(this.getExtension(), ext);
```

Comment:
```suggestion
        return this.getTextFileName().slice(0, -this.getExtension().length) + ext;
```

or something like that; this will fail on converting `grumpy.py` / `knots.ts`

Reply:
yep